### PR TITLE
ENH: Update azureml packages versions. 

### LINKED ISSUE #475: Many warnings are shown in the training logs of histo pipelines.
We can hardly read the training logs of histopathology pipelines because of some azure dependency warnings.

```text
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (msal-extensions 1.0.0 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('msal-extensions~=0.3.0'), {'azure-identity'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (msal-extensions 1.0.0 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('msal-extensions~=0.3.0'), {'azure-identity'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint automl = azureml.train.automl.run:AutoMLRun._from_run_dto with exception (msal-extensions 1.0.0 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('msal-extensions~=0.3.0'), {'azure-identity'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.scriptrun = azureml.core.script_run:ScriptRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.PipelineRun = azureml.pipeline.core.run:PipelineRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.ReusedStepRun = azureml.pipeline.core.run:StepRun._from_reused_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint azureml.StepRun = azureml.pipeline.core.run:StepRun._from_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
Failure while loading azureml_run_type_providers. Failed to load entrypoint hyperdrive = azureml.train.hyperdrive:HyperDriveRun._from_run_dto with exception (azure-core 1.22.1 (/azureml-envs/azureml_ad828c522843e7a07e5ccbe50abb28a8/lib/python3.7/site-packages), Requirement.parse('azure-core<2.0.0,>=1.23.0'), {'azure-mgmt-core'}).
```

### PR DESCRIPTION
**Pull request by kenza-bouzid**
Closes #475

### CODE REVIEW

File: hi-ml-histopathology/src/histopathology/utils/tiles_selection_utils.py [hunk 1]
```diff
@@ -67,7 +67,6 @@ def update_selected_tiles(self, tiles: Tensor, attn_scores: Tensor, num_top_tile
         :param attn_scores: A tensor of attention scores assigned by the deepmil model to tiles of shape (n_tiles, 1).
         :param num_top_tiles: The number of tiles to select as top and bottom tiles.
         """
-        print("attn_scores.shape", attn_scores.shape)
```
Comment (kenza-bouzid) [inline hi-ml-histopathology/src/histopathology/utils/tiles_selection_utils.py:70]: My bad, I merged some debugging prints as part of  #465

**Review by ant0nsc — APPROVED** [review]

**Comment by mebristo** [comment]
Coudl you check that the issue [described here](https://github.com/microsoft/hi-ml/issues/421) is solved in this azureml package version before merging please? In which case we may be able to unpin `msrest`

**Comment by kenza-bouzid** [comment]
> Coudl you check that the issue [described here](https://github.com/microsoft/hi-ml/issues/421) is solved in this azureml package version before merging please? In which case we may be able to unpin `msrest`

It seems like it solved it. azureml-core 1.43.0 has an upper bound for azure-core <2.0. Now the env has azure-core 1.24.1. I submitted some jobs to AML to confirm it, but it seems like we can safely unpin `msrest`.

**Comment by mebristo** [comment]
> > Coudl you check that the issue [described here](https://github.com/microsoft/hi-ml/issues/421) is solved in this azureml package version before merging please? In which case we may be able to unpin `msrest`
> 
> It seems like it solved. azureml-core 1.43.0 has an upper bound for azure-core <2.0. Now the env has azure-core 1.24.1. I submitted some jobs to AML to confirm it, but it seems like we can safely unpin `msrest`.

Thanks!

File: hi-ml-histopathology/Makefile [hunk 2]
```diff
@@ -6,6 +6,10 @@ env:
 	pip install -r requirements_test.txt
 	pip install -r requirements_build.txt
 
+# Update the Conda environment
+conda_update:
+	conda env update -n $(CONDA_DEFAULT_ENV) --file environment.yml
```
Comment (mebristo) [inline hi-ml-histopathology/Makefile:11]: Is it worth adding `--prune` here to remove packages that aren't part of the environment definition? E.g. if we remove some dependencies

**Review by mebristo — APPROVED** [review]

File: hi-ml-histopathology/Makefile (hunk 2 above)
Comment (kenza-bouzid) [reply hi-ml-histopathology/Makefile:11]: Good point!

**Pull request merged**